src/volume_calc.cpp

```diff
@@ -59,7 +59,39 @@ VolumeCalculation::VolumeCalculation(pugi::xml_node node)
   domain_ids_ = get_node_array<int>(node, "domain_ids");
   lower_left_ = get_node_array<double>(node, "lower_left");
   upper_right_ = get_node_array<double>(node, "upper_right");
-  n_samples_ = std::stoi(get_node_value(node, "samples"));
+  std::stringstream size_t_stream(get_node_value(node, "samples"));
```

Comment:
Wouldn't `std::stoull` be sufficient?

Reply:
I guess this is a bit more general, but also quite verbose and probably not necessary. Replacing...